### DISCUSSION TITLE
How to add proto_path to protoc call with nanopb integration

### DISCUSSION THREAD

**Discussion by asteriskSF** [zephyrproject-rtos/zephyr#78360] [General]
@pdgendt you might be able to help me with this one as well!

I have some common protobuf message definitions in one directory and in another parallel directory my a.proto includes the comon protobuf file.  
I would like to  add a --proto_path argument to the invokation of protoc.  

directory structure
top/
top/common/common_defs.proto
top/lib/lib.proto      #has import "common_defs.proto"

In the CMakeLists.txt within top/lib/    I tried various things such as:
set(PROTOBUF_IMPORT_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../common)
set(PROTOBUF_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../common)
set(Protobuf_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../common)
set(Protobuf_IMPORT_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/../common)
set(PROTOBUF_OPTIONS "--proto_path=${CMAKE_CURRENT_SOURCE_DIR}/../common")
include_directories(${Protobuf_INCLUDE_DIRS})

but I couldn't get it to work.

also I tried a relative path in the import, but protoc does not allow that.

**Comment by asteriskSF** [comment]
BTW,  I'm on cmake 3.21.

Is the only solution to dump all the files into a single directory?

**Reply by pdgendt** [reply]
It will save you a lot of trouble if you keep your proto files together IMO.

**Comment by asteriskSF** [comment]
The solution I chose was to build protobuf  from the parent directory with hierarchical descending filenames.  The existing code required some updates but they were pretty simple.  This method avoided moving the existing protobuf files from their modular folders into a single common folder.
